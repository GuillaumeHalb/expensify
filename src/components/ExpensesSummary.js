import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import selectExpensesTotal from "../selectors/expenses-total";
import selectExpense from "../selectors/expenses";

export const ExpensesSummary = props => (
  <div>
    {props.expenses.length === 0 ? (
      <p>select some expenses</p>
    ) : (
      <p>
        {`Viewing ${props.expenses.length} expense${
          props.expenses.length > 1 ? "s" : ""
        } totalling
          ${numeral(selectExpensesTotal(props.expenses) / 100).format(
            "$0,0.00"
          )}`}
      </p>
    )}
  </div>
);

const mapStateToProps = state => ({
  expenses: selectExpense(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);
