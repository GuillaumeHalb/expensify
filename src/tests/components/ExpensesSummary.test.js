import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import EXPENSES from "../fixtures/expenses";

test("should render ExpensesSummary with expenses", () => {
  const wrapper = shallow(<ExpensesSummary expenses={EXPENSES} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpensesSummary with empty message", () => {
  const wrapper = shallow(<ExpensesSummary expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
