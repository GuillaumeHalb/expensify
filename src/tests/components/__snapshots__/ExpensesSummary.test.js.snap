// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with empty message 1`] = `
<div>
  <p>
    select some expenses
  </p>
</div>
`;

exports[`should render ExpensesSummary with expenses 1`] = `
<div>
  <p>
    Viewing 3 expenses totalling
              $1,141.95
  </p>
</div>
`;
